app/grandchallenge/cases/templates/cases/image-search-widget.html

```diff
@@ -0,0 +1,30 @@
+{% load url %}
+{% load static %}
+
+<div class="form-group mb-1"
+     id="search-image-{{ widget.name }}"
+     hx-get="{% url 'cases:image-search' %}"
+     hx-trigger="load once, keyup[target.value.length > 1] delay:500ms from:#query-{{ widget.name }}, click[keyCode==8] delay:500ms from:#query-{{ widget.name }}"
+     hx-target="#selected-image-{{ widget.name }}"
+     hx-swap="outerHTML"
+     hx-vals='{"interface": "{{ widget.name }}"}'
+     hx-include="[id='query-{{ widget.name }}']"

```

Comment:
I'm still curious about the need for `hx-vals` (both here and other places) and `hx-include`, but this comes from my lack of knowledge about htmx. Are they really necessary? Why aren't these things being sent along with a form?

Reply:
`hx-include` is used to include additional element values in the request (from elements outside of the form that the htmx is configured for). Values from elements within the form are sent along automatically. Looking at this case here again, it's actually redudant! It's also the only place in the gc repo where we use this at the moment. I'll remove it. I believe in the gc-private repo it was necessary everytime I used it, but I'll double check that assumption when I have a moment.

`hx-vals` is used to add extra parameters to a request that are not coming from other elements in the DOM. I think the only other way of sending the interface slug along would be to add a hidden input element to the form, but I don't see how that would be any better than passing it through hx-vals in this particular case where we control the value (it's not user generated).

Reply:
Ah so for the `hx-include`: it turns out to be necessary because I'm not using a form element, but a div here. Using a form is not possible, because then we'd have a form nested in a form, which leads to issues with the final form submission.

Reply:
> I think the only other way of sending the interface slug along would be to add a hidden input element to the form

But what about the case above with the help text? We allow the user to customise that in some places. If we think some values are safe now it is very difficult to keep track of all those usages, IMO it is better to be conservative and use hidden form elements.

Reply:
Fair enough. I have added hidden elements and include them with `hx-include`